spec/src/main/asciidoc/config.asciidoc

```diff
@@ -0,0 +1,147 @@
+//
+// SPDX-License-Identifier: Apache-2.0
+
+[[config]]
+== Common Configuration Patterns
+
+Many MicroProfile specifications make use of the Configuration specification for
+setting or overriding behaviour characteristics. Listed below are a list of recommended usage
+patterns that are already adopted by one or more of the MicroProfile specifications.
+
+[[patterns]] 
+=== Patterns
+
+[[propertyPrefix]]
+==== Configuration Property Name Prefixes
+
+In order to prevent collision it is recommended that properties that are not
+named to match application class names are prefixed with "mp." followed by the specification's short name:  `mp.<spec-name>`.
+
+----
+// For example, the Messaging Specification has an @Outgoing annotation:
+   mp.messaging.outgoing.attributename=3
+----
+
+[[qualifyingClassNames]]
+==== Properties That Relate To Application Classes
+
+All references to users' application classes in property names should be fully package qualified.
+
+There are places where an annotation class name appears or a specification sourced class name (e.g. Context Propagation) appears in configuration property names, these class names are not package qualified. Should these start with `mp.<spec-name>`? In this case `mp.ft`?
+----
+// For example: 
+   Retry/maxRetries=30
+//or should this be?
+   mp.ft.retry.maxRetries=30
+----
+This is probably the right choice from a usability point of view. 
+This is particularly not a problem where the scope of the configuration property identifies a particular Java file (which will have resolved the package name via an import).
+It could become more of a problem for service wide defaults if more than one MicroProfile service
+has a class with the same name. 
+The result is that we should have exclusive class names in MicroProfile for classes that might need configuration override in the process environment. 
+The 'first' specification that owns a class name should own that name exclusively. 
+
+[[fieldAnnotations]]
+==== Properties That Relate To Annotated Fields
+
+It is possible to override attributes of annotated fields. In this case the following
+pattern is recommended: `<fully qualified class name>/<field name>/<spec-defined annotation name>/<annotation attribute name>`.
+----
+// For example:
+   com.acme.ClassA/fieldVat/ConfigProperty/defaultValue=20
+----
+
+[[methodAnnotations]]
+==== Properties That Relate To Annotated Methods
+
+For annotated application methods, these are expected to be overridden
+with the following pattern: `<fully qualified class name>/<method name>/<spec-defined annotation short name>/<annotation attribute name>`.
+----
+// For example:
+   com.acme.ClassA/methodB/Retry/maxRetries=5
+----
+
+[[overloadedMethods]]
+==== Properties That Relate To Overloaded Methods
+
+There are places in the configuration where a method name is used. 
+Accounting for methods with the same name but different signatures
+does not have a recommended approach yet.
```

Comment:
There may also be clash between field and method name. I am not sure if we have any annotations that can be applied to both yet but it should be pointed out.

Reply:
Metrics has annotations that can be applied to both. But currently nothing in that area that could be configured via MPConfig(-properties)

Reply:
I will remove such 'chatty' text into a seperate issue/PR with another commit and resolve this conversation point when this is done.

Reply:
Done, see https://github.com/eclipse/microprofile/issues/118